### ISSUE TITLE
Feature: Clear the list of files

### ISSUE THREAD

**Issue by mat-u** [ImageOptim/ImageOptim#188]
When optimizing many files, and re-adding other ones, the list can be huge (scroll). This can be difficult to read which files are ok or not when working by lot.
Suggestions, when all files are optimized, add a "Clear" button to delete the list near "Again" (disabled when a file is processing).
Currently, the only solution is to close the application, or I'm missing a way.

Your thought?

**Comment by pathawks** [comment]
Cmd-A Delete

**Comment by pornel** [comment]
Yes, I use the shortcut, but this issue keeps coming back, so let's try to solve it.

I'm not keen on the specific solution of adding another button. If possible I'd prefer something that "just works" automatically rather than add more things for user to click or configure.

How about clearing the list when you drop new files? Or perhaps when you drop more than one file, or if the list is longer than 100 files, or if files have been on the list for longer than a minute? 

How about removing the list completely? Just have a window with a drop area and a progress bar. The list could be another window if you click a "Details" button or such.

Or how about using an outline view (like Finder's folder view with triangles next to folders), where each optimization/drop of files is a "folder" and only the latest one is expanded?

**Comment by curby** [comment]
> How about clearing the list when you drop new files? Or perhaps when you drop more than one file, or if the list is longer than 100 files, or if files have been on the list for longer than a minute?

For any of these, please only clear files that were successfully processed, i.e. if there was an error, then leave them in the list.  

I kind of like the folders idea though, so the results of each batch of files can be tracked.  Not sure what's the best way to represent the jobs/batches.  Handbrake's job queue used to do something similar; might be worth looking at for ideas.  

Incidentally, what's wrong with the button idea?  Clutter?  It seems to make sense thematically: those buttons would consistently act upon the items in the list: remove items, or process them again.  I'm not really campaigning for another button, but am curious what your reasoning is.

**Comment by pathawks** [comment]
As someone who frequently uses the “again” button, clearing files from the list before I’m ready sounds a bit scary.

**Comment by pornel** [comment]
Regarding more buttons, I have two objections:

- Philosophical. ImageOptim still has the same number of buttons as it had almost 10 years go. If I kept adding "just one more" button it'd probably have a dozen now. I'm sticking to the principle that it's a simple single-purpose app that won't do everything.

- Practical. I don't think there's enough room for it. The status bar needs a lot of space, otherwise it looks cluttered.

I could probably switch the layout to fat titlebar, something more like Safari/Calendar. That would have room for 3-4 buttons.

**Comment by mat-u** [comment]
@pornel I agree for the simplicity of the UI of ImageOptim. Maybe just adding a bold line, or a visuel guide between each lot will be a good solution, when droping/adding new files?

Example:

![example](https://cloud.githubusercontent.com/assets/2226408/23605278/47905752-025c-11e7-9b1f-22d3d8d12307.png)

As a side reasoning to simplify (maybe a new issue?), why don't labelize the "+" with a button"+ Add" or "+ Add files" to be more consistent, and remove the text "Drag and drop image files onto the area above" which can be explicit with the drop zone and the "Add file". If I can have your thought, I'm not pretending revolutionize the app.

![imageoptim_ui](https://cloud.githubusercontent.com/assets/2226408/23605872/ac28f82a-025e-11e7-9c5b-011e6f103e45.png)

(Sorry for the screenshot in french, and if my english is not good :) )

**Comment by pornel** [comment]
Unfortunately I can't add the line, because the list is sortable.

**Comment by curby** [comment]
Great discussion here!  I'm a fan of the fat titlebar.  Among the many benefits:

1. You can customize which buttons to show, and how to show them (icons/labels/both).
2. You can hide the toolbar entirely for an even cleaner view.
3. The statusbar is turned back into a status-only bar instead of its current hybrid role with buttons.
4. It better follows macOS UI conventions and Apple apps like Finder.

Of course, batch-based organization, i.e. in folders etc., might still be an interesting idea independent of toolbar design.

**Comment by JoeDawson** [comment]
I too would like to see this implemented. Perhaps in a similar way to how [Handbrake](https://github.com/HandBrake/HandBrake) has an option to remove an item from the queue.

![screen shot 2017-03-30 at 10 35 29](https://cloud.githubusercontent.com/assets/1009696/24497956/028d4bd8-1535-11e7-80fb-1e01fd8095af.jpg)

Let's say you add a batch of 5 items to ImageOptim and you're satisfied with 3 of then, you can remove the 3 leaving the remaining 2 to run again.

Just adding my thoughts, I would like a "Clear All" option too though. I rarely run ImageOptim "Again" and just move onto my next lot of images to compress.

**Comment by mrleblanc101** [comment]
Could you make so that image disappear from the list X seconds after being optimized ?
